src/components/structures/EmbeddedPage.js

```diff
@@ -82,19 +84,31 @@ export default class EmbeddedPage extends React.PureComponent {
                 this.setState({ page: body });
             },
         );
+
+        this._dispatcherRef = dis.register(this.onAction);
     }
 
     componentWillUnmount() {
         this._unmounted = true;
+        dis.unregister(this._dispatcherRef);
     }
 
+    onAction = (payload) => {
+        // HACK: Workaround for the context's MatrixClient not being set up at render time.
+        if (payload.action === 'client_started') {
+            this.forceUpdate();
+        }
+    };
+
     render() {
-        const client = this.context.matrixClient;
+        // HACK: Workaround for the context's MatrixClient not updating.
+        const client = this.context.matrixClient || MatrixClientPeg.get();
```

Comment:
Yeah, if it's easier, feel free to not use the context here.

Reply:
I think I'll leave it for compatibility concerns, and next time I swing by the component I'll take an axe to it.